week1/plot_trips.R

```diff
@@ -1,62 +1,154 @@
+# hint: try using the pivot_longer() function for this to reshape things before plotting
+# (you can skip this and come back to it tomorrow if we haven't covered reshaping data yet)
+
+gather(weather, "temp", "tnums", 5:6) %>% 
+  ggplot(aes(x=date, y=tnums, color=temp)) + 
+  geom_point()
+
+########################################
+# plot trip and weather data
+########################################
+
+# join trips and weather
+trips_with_weather <- inner_join(trips, weather, by="ymd")
+
+# needed to modify weather to get ymd as a date 
+weather_ymd <- weather %>% mutate(ymd = as.Date(date, '%Y-%m-%d'))
+
+trips_with_weather <- inner_join(trips, weather_ymd, by="ymd")
+weather <- weather_ymd
+
+# plot the number of trips as a function of the minimum temperature, where each point represents a day
+# you'll need to summarize the trips and join to the weather data to do this
+num_trips <- trips %>% group_by(ymd) %>% summarize(num_trips = n())
+num_trips_tmin <- inner_join(num_trips, weather, by="ymd")
+
+num_trips_tmin %>% 
+  ggplot(mapping = aes(x = tmin, y = num_trips)) +
+  geom_point() +
+  scale_y_continuous(label = comma)
+
+# repeat this, splitting results by whether there was substantial precipitation or not
+# you'll need to decide what constitutes "substantial precipitation" and create a new T/F column to indicate this
+num_trips_tmin %>% 
+  mutate(sub_precip = prcp >= 0.3) %>%
+  ggplot(mapping = aes(x = tmin, y = num_trips, color = sub_precip)) +
+  geom_point() +
+  scale_y_continuous(label = comma)
+
+# add a smoothed fit on top of the previous plot, using geom_smooth
+num_trips_tmin %>% 
+  mutate(sub_precip = prcp >= 0.3) %>%
+  ggplot(mapping = aes(x = tmin, y = num_trips, color = sub_precip)) +
+  geom_point() +
+  geom_smooth() +
+  scale_y_continuous(label = comma)
+
+# compute the average number of trips and standard deviation in number of trips by hour of the day
+# hint: use the hour() function from the lubridate package
+
+trips_with_weather$hour <-hour(trips_with_weather$starttime)
+
+summ_by_hour <- trips_with_weather %>% 
+  group_by(ymd, hour) %>%
+  summarize(count = n())
+
+hours_summary <-group_by(summ_by_hour, hour) %>% 
+  summarize(average_num_trips = mean(count), 
+            sd_num_trips = sd(count))
+
+
+# plot the above
+
+# repeat this, but now split the results by day of the week (Monday, Tuesday, ...) or weekday vs. weekend days
+# hint: use the wday() function from the lubridate package
+
+trips_with_weather %>%
+  mutate(hour = hour(starttime), wday = wday(starttime)) %>%
+  group_by(ymd, hour, wday) %>%
+  summarize(num_trips = n()) %>%
+  group_by(hour, wday) %>%
+  summarize(average_trips = mean(num_trips),
+            sd_trips = sd(num_trips)) %>%
+  ggplot(mapping = aes(x = hour)) +
+  geom_point(mapping = aes(y = average_trips, color = 'avg')) +
+  geom_point(mapping = aes(y = sd_trips, color = 'sd')) +

```

Comment:
looks pretty good, I probably would've used a line here though, since when the quantities vary a lot over the course of the day, then scatter plots can get hard to read. Plus you don't have to worry about collisions on the X-Axis because you already grouped by hour.